contracts/Exchange.sol

```diff
@@ -154,22 +152,32 @@ contract Exchange is Restricted {
         return(activeBuyOrders.length, activeSellOrders.length);
     }
 
-    // returns CHUNK_SIZE orders starting from offset
+    // returns <chunkSize> active buy orders starting from <offset>
     // orders are encoded as [id, maker, price, amount]
-    function getActiveBuyOrders(uint offset) external view returns (uint[4][CHUNK_SIZE] response) {
-        for (uint8 i = 0; i < CHUNK_SIZE && i + offset < activeBuyOrders.length; i++) {
-            uint64 orderId = activeBuyOrders[offset + i];
+    function getActiveBuyOrders(uint offset, uint16 chunkSize)
+    external view returns (uint[4][]) {
+        uint[4][] memory response = new uint[4][](chunkSize);
```

Comment:
You might slightly optimize memory footprint here:
`new uint[4][](<correct size>);` requires less than `chunkSize` at the end.

probably best solution is just to carefully choose chunkSize

Reply:
Good point, thanks.